### ISSUE TITLE
Table Component:  showCheckbox, expansion will not work if rows is initialized empty

### ISSUE THREAD

**Issue by mechevarria** [patternfly/patternfly-ng#337]
If loading via an observable and having

 `rows=[]`

having the array populating by an Observable later, like loading remote data and you have `showCheckbox=true`, the checkboxes aren't shown (blank cell)
This also affects the `expansion` option as well.

I have a workaround example here

https://github.com/mechevarria/patternfly-ng-seed/blob/master/src/app/table/table.component.ts

I initialize 

`rows: Comment[] = [new Comment()];`

In other words, to a non-null object.  Even `{}` will work.  Then in `ngAfterViewInit()` I set `rows = []`.  That allows the table to be populated from an Observable and have the checkboxes work.

In the example, if I change

`rows: Comment[] = [new Comment()];`

to

`rows: any[] = [];`

and change the **TableConfig** to

`showCheckbox: true`

then that will break the showing of checkboxes (blank cell at beginning of row)

**Comment by mechevarria** [comment]
![no-checkbox](https://user-images.githubusercontent.com/2657771/39210780-b924a5be-47d7-11e8-9e0e-193374d25c5c.png)

This is an example with data loaded remotely

**Comment by dlabrecq** [comment]
Have you tried forcing a change detection described here?
https://swimlane.gitbooks.io/ngx-datatable/introduction/cd.html

`this.rows = [...this.rows];`

We also have a examples of an empty table below. Choose either the "Embedded View" or "Table View" examples, then click "Rows Available". This forces a change detection as described above when populating an empty table with data.

http://www.patternfly.org/patternfly-ng/#/table

**Comment by mechevarria** [comment]
Shoot.  I did read that and didn't connect the dots.  Let me try that

**Comment by mechevarria** [comment]
hmm.  Seems like its still a problem.  I can blank the array the after the page initializes with no issue, so `this.rows = []` and then re-populate with data and be able to have checkboxes.

but `this.rows` has to be an array of an object on view init, it can't be a blank array on view init.  The table example starts with data, not a blank array so it will work.

**Comment by dlabrecq** [comment]
Ok, thanks for trying that. I'll try modifying an example to start with no data.

**Comment by dlabrecq** [comment]
I was able to reproduce the issue, but could not apply your workaround directly to the table component. Adding a dummy object to the rows input will render an empty row, which is not desirable.

That said, I did find another fix that works well for the table component and created the PR below. Thanks for pointing us in the right direction.

https://github.com/patternfly/patternfly-ng/pull/338

**Comment by mechevarria** [comment]
Totally agree about the empty row.  My hack around that was to clear out the array in `ngAfterViewInit()`, which is also not desirable.  Glad a fix was found so fast, appreciate it.
